android-components/components/support/utils/src/test/java/mozilla/components/support/utils/URLStringUtilsTest.kt

```diff
@@ -88,6 +88,40 @@ class URLStringUtilsTest {
         assertFalse(isURLLike("www.c-c-  "))
         assertFalse(isURLLike("3-3 "))
 
+        // IPv6 literals
+        assertTrue(isURLLike("[::]"))
+        assertTrue(isURLLike("[::1]"))
+        assertTrue(isURLLike("[1::]"))
+        assertTrue(isURLLike("[::]/"))
+        assertTrue(isURLLike("[1:2:3:4:5:6:7:8]"))
+        assertTrue(isURLLike("https:[::]"))
+        assertTrue(isURLLike("https://[::]"))
+        assertTrue(isURLLike("https://[::1]"))
+        assertTrue(isURLLike("https://[1::]"))
+        assertTrue(isURLLike("https://[2001:db8::1.2.3.4]/"))
+        assertTrue(isURLLike("https://[1:2:3:4:5:6:7:8]/"))
+        assertTrue(isURLLike("[2001:db8::1.2.3.4]/"))
```

Comment:
Here is a suggestion that I think makes it easier to test several variations of each "base IPv6 literal", including `http:`, `https:`, trailing slash... 
```suggestion
        val validBaseUrls = listOf(
            "[::]",
            "[::1]",
            "[1::]",
            "[::]/",
            "[1:2:3:4:5:6:7:8]",
            "[2001:db8::1.2.3.4]",
        )

        validBaseUrls.forEach { url ->
            assertTrue(isURLLike(url))
            assertTrue(isURLLike("$url/"))
            assertTrue(isURLLike("https://$url"))
            assertTrue(isURLLike("https://$url/"))
            assertTrue(isURLLike("http://$url"))
            assertTrue(isURLLike("http://$url/"))
        }
```

What do you think? 

Reply:
Done.  I removed "[::]/" because the / is tested later, added "[::1]:8080" for coverage, and added tests for "https:[::]"-style URLs.

Reply:
Good catch!